plugins/CoreHome/vue/src/MatomoCopyModal/MatomoCopyModal.vue

```diff
@@ -0,0 +1,330 @@
+import { MatomoCopyModalStore } from './MatomoCopyModalStore';
+
+// async since we're referencing a recursive component
+const Field = useExternalPluginComponent('CorePluginsAdmin', 'Field');
+const Form = useExternalPluginComponent('CorePluginsAdmin', 'Form');
+
+const { $ } = window;
+
+interface MatomoCopyModalState {
+  isLoading: boolean;
+  isValidated: boolean;
+  descriptionLearnMoreLink: string;
+  copyErrors: string[];
+  site: SiteRef|null;
+  hasSiteBeenInitialised: boolean;
+  hasBeenSubmitted: boolean;
+}
+
+interface CopyRequestResponse {
+  isCopySuccessful?: boolean;
+  successMessage?: string;
+  responseData?: Record<string, unknown>;
+  errorMessage?: string;
+  errorCode?: number;
+}
+
+export default defineComponent({
+  directives: {
+    Form,
+  },
+  components: {
+    Field,
+  },
+  props: {
+    /**
+     * The reactive class for controlling the settings of the modal from multiple components.
+     */
+    modalStore: {
+      type: MatomoCopyModalStore,
+      required: true,
+    },
+  },
+  data(): MatomoCopyModalState {
+    return {
+      isLoading: true,
+      isValidated: false,
+      descriptionLearnMoreLink: '',
+      copyErrors: [],
+      site: null,
+      hasSiteBeenInitialised: false,
+      hasBeenSubmitted: false,
+    };
+  },
+  emits: [
+    'copySuccessful',
+    'copyFailed',
+  ],
+  watch: {
+    isModalVisible(newValue) {
+      if (!newValue) {
+        return;
+      }
+
+      // TODO - Do some logic before showing modal
+
+      this.showCopyModal();
+
+      // TODO - determine the best indication that loading is done
+      this.isLoading = false;
+    },
+    site() {
+      this.onSiteChange();
+    },
+  },
+  methods: {
+    closeModal() {
+      const root = this.$refs.root as HTMLElement;
+      const $root = $(root);
+      $root.modal('close');
+    },
+    resetModal() {
+      this.modalStore.hideModal();
+      this.site = null;
+      this.isLoading = true;
+      this.isValidated = false;
+      this.copyErrors = [];
+      this.hasSiteBeenInitialised = false;
+      this.hasBeenSubmitted = false;
+    },
+    showCopyModal() {
+      const root = this.$refs.root as HTMLElement;
+      const $root = $(root);
+      $root.modal({
+        dismissible: true,
+        onCloseEnd: () => {
+          this.resetModal();
+        },
+      }).modal('open');
+    },
+    submitCopy() {
+      this.hasBeenSubmitted = true;
+      this.modalStore.disableWatchSuppression();
+
+      // Make sure all the validation passes before making the server request
+      this.validateFormFields();
+      if (!this.getIsValid) {
+        this.hasBeenSubmitted = false;
+        return;
+      }
+
+      // Actually POST the API call
+      const ajax = new AjaxHelper();
+      // Remove the unnecessary default parameters
+      ajax.removeDefaultParameter('date');
+      ajax.removeDefaultParameter('period');
+      ajax.removeDefaultParameter('segment');
```

Comment:
Could perhaps have a list defined in a constant so that it can be extended if needed instead of adding more lines here? Alternatively, does it cause any problems leaving the params there?